src/storage/query/QueryBaseProcessor.inl

```diff
@@ -456,6 +564,19 @@ kvstore::ResultCode QueryBaseProcessor<REQ, RESP>::collectEdgeProps(
                                                   val,
                                                   spaceId_,
                                                   std::abs(edgeType));
+
+            // Check if the schema has TTL
+            if (edgeHasTTL(edgeType)) {
+                std::pair<std::string, int64_t> edgeTTL = getEdgeTTLInfo(edgeType);
```

Comment:
Move getEdgeTTLInfo outside of the for loop.

Reply:
Here is a check of the value of each row